xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java

```diff
@@ -273,6 +275,31 @@ private void handleEndpointResolutionError() {
       }
     }
 
+    /**
+     * Wires re-resolution requests from downstream LB policies with DNS resolver.
+     */
+    private final class RefreshableHelper extends ForwardingLoadBalancerHelper {
+      private final Helper delegate;
+
+      private RefreshableHelper(Helper delegate) {
+        this.delegate = checkNotNull(delegate, "delegate");
+      }
+
+      @Override
+      public void refreshNameResolution() {
+        for (ClusterState state : clusterStates.values()) {
+          if (state instanceof LogicalDnsClusterState) {
+            ((LogicalDnsClusterState) state).refresh();
+          }
+        }
```

Comment:
Also call `delegate. refreshNameResolution()`?

Reply:
Hmm... I don't think so. We can think of this wrap as a complete _override_ instead of an extension.

From the high-level design view, we shouldn't. It does nothing even if we call `delegate.refreshNameResolution()` as this is how xDS works (async resolver). Previously we did not explicitly override this API because to bypass re-resolution requests to the xDS resolver  since we know it does nothing. Not bother write code to turn some operation into no-op given that it's already a no-op. Same would apply here, not bother explicitly add operations that known to be no=op.